src/library/dlgcoverartfullsize.cpp

```diff
@@ -28,14 +31,17 @@ DlgCoverArtFullSize::DlgCoverArtFullSize(QWidget* parent, BaseTrackPlayer* pPlay
             &DlgCoverArtFullSize::customContextMenuRequested,
             this,
             &DlgCoverArtFullSize::slotCoverMenu);
-    connect(m_pCoverMenu,
-            &WCoverArtMenu::coverInfoSelected,
-            this,
-            &DlgCoverArtFullSize::slotCoverInfoSelected);
-    connect(m_pCoverMenu,
-            &WCoverArtMenu::reloadCoverArt,
-            this,
-            &DlgCoverArtFullSize::slotReloadCoverArt);
+    m_pWCoverArtMenu = dynamic_cast<WCoverArtMenu*>(pCoverMenu);
```

Comment:
I think it is better to use here a common base class that has only the signals.

Reply:
So Instead of QMenu should I use WCoverArtMenu in this case? For now I think only WCoverArtMenu can work so we can continue with the fetcher itself. A menu for fetcher can be implemented later on IMHO.

Wouldn't QObject casting work? Since it doesn't use RTTI, wouldn't it be faster and solve the issue regarding with the dynamic_casting? 

Or the problems with the casting it is not only the performance is it also technical stuff too?

Reply:
Yes technical, a QOject cast will also work and should indeed be preferred.

The main issue with upcasting, is that the expected child class is hidden in the implementation of the function body.

Reply:
If i understand it correctly, we should go with the WCoverArtMenu instead of QMenu in the constructor right?

Reply:
Yes, that was my idea. We can alter the code later again if we introduce a second menu class.

Reply:
Thank you for explaining, that's fair enough in the first place as you mentioned, with the second menu class it can be changed :+1:

Reply:
Done :+1: I tested it with label without menu, it worked fine.